components/net/blob_loader.rs

```diff
@@ -0,0 +1,91 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+use hyper::header::{DispositionType, ContentDisposition, DispositionParam};
+use hyper::header::{Headers, ContentType, ContentLength, Charset};
+use hyper::http::RawStatus;
+use ipc_channel::ipc::{self, IpcSender};
+use mime::{Mime, Attr};
+use mime_classifier::MIMEClassifier;
+use net_traits::ProgressMsg::Done;
+use net_traits::blob_url_store::{parse_blob_url, BlobURLStoreEntry, BlobURLStoreError, BlobURLStoreMsg};
+use net_traits::response::HttpsState;
+use net_traits::{LoadConsumer, LoadData, Metadata, NetworkError};
+use resource_thread::{CancellationListener, send_error, start_sending_sniffed_opt};
+use std::str;
+use std::sync::Arc;
+
+
+// TODO: Check on GET
+// https://w3c.github.io/FileAPI/#requestResponseModel
+
+pub fn factory(load_data: LoadData,
```

Comment:
I don't think this is how the factories work. This function is more like load_for_consumer in http_loader, the factory is a function that returns functions that call load_for_consumer.

Additionally, there should be a thread involved here.